kubespawner/spawner.py

```diff
@@ -106,7 +106,7 @@ def events(self):
         #   timestamp without and the other is a higher resolution timestamp.
         return sorted(
             self.resources.values(),
-            key=lambda event: event.last_timestamp or event.event_time,
+            key=lambda event: event["lastTimestamp"] or event["eventTime"],
```

Comment:
This is a good example of the kind of issue I'm talking about above. Handling this for the known event and pod reflectors within this repo is sufficient but I worry about breaking out of tree reflectors, but like I said I also don't know what contract is there.